lib/bandit/http2/connection.ex

```diff
@@ -513,4 +529,40 @@ defmodule Bandit.HTTP2.Connection do
   defp send_frame(frame, socket, connection) do
     Socket.send(socket, Frame.serialize(frame, connection.remote_settings.max_frame_size))
   end
+
+  defp handle_initial_request({method, request_target, headers, body}, socket, connection) do
+    {_, _, _, path} = request_target
+    headers = [{":scheme", "http"}, {":method", method}, {":path", path} | headers]
+
+    with {:ok, stream} <- StreamCollection.get_stream(connection.streams, 1),
+         true <- accept_stream?(connection),
+         true <- accept_headers?(headers, connection.opts, stream),
+         {:ok, stream} <-
+           Stream.recv_headers(
+             stream,
+             connection.transport_info,
+             connection.telemetry_span,
+             headers,
+             true,
+             connection.plug,
+             body,
+             connection.opts
+           ),
+         stream = %{stream | state: :remote_closed},
+         {:ok, streams} <- StreamCollection.put_stream(connection.streams, stream) do
+      {:ok, %{connection | streams: streams}}
```

Comment:
I wonder if we shouldn't be DRYing this up with the existing header receiving logic in the `handle_frame(%Header{})` function above (basically have `handle_frame(%Header{})` handle the header decompression only, and pull most of that function out to an internal `handle_headers()` function that we could reuse here).

Reply:
Addressed in b8cf932